docroot/sites/all/modules/dev/warmshowers_site/warmshowers_site.module

```diff
@@ -89,39 +89,20 @@ function warmshowers_site_theme() {
  * @return mixed
  */
 function warmshowers_site_preprocess(&$variables, $key) {
-  global $user;
-  if ($user) {
-    $variables['account'] = $user;
-  }
-
-  _warmshowers_site_get_theme_variables($key, $variables);
-}
-
-/**
- * Helper function to set default variables for theme functions.
- *
- * @param string $delta
- *  The delta name for the theme function being called.
- * @param array $variables
- *  An array of variables to be modified or appended to.
- *
- * @return array
- */
-function _warmshowers_site_get_theme_variables($delta, &$variables) {
 
-  switch ($delta) {
+  switch ($key) {
     case 'ws_user_account_menu':
-      if ($variables['account'] && !empty($variables['account']->uid)) {
-        $variables['fullname'] = $variables['account']->data['fullname'];
-        $variables['profile_link'] = isset($variables['account']->data['fullname']) ? l($variables['account']->data['fullname'], 'user/' . $variables['account']->uid) : '';
+      if (!empty($GLOBALS['user']->uid)) {
+        $account = user_load($GLOBALS['user']->uid);
+        $variables['profile_link'] = theme('username', array('account' => $account));
```

Comment:
theme('username') will return rfay not Randy Fay unless you change that in hook theme no??